### PR TITLE
Removed the ? prefix from the QUERY_STRING server variable

### PR DESCRIPTION
**Pull request by mikaelm12**
Addresses https://github.com/aspnet/BasicMiddleware/issues/148
@Tratcher @BrennanConroy

### CODE REVIEW

File: src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ public class QueryStringSegment : PatternSegment
     {
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return context.HttpContext.Request.QueryString.ToString();
+            return context.HttpContext.Request.QueryString.ToString().Substring(1);
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs]: Throws ArgumentOutOfRangeException if there is no query

File: src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs [hunk 2]
```diff
@@ -7,7 +7,14 @@ public class QueryStringSegment : PatternSegment
     {
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return context.HttpContext.Request.QueryString.ToString();
+            var queryString = context.HttpContext.Request.QueryString.ToString();
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs]: Just to be sure queryString can never be null, right? If so you can probably use IsEmpty instead/

File: test/Microsoft.AspNetCore.Rewrite.Tests/PatternSegments/QueryStringSegmentTests.cs [hunk 3]
```diff
@@ -12,16 +12,13 @@ public class QueryStringSegmentTests
         [Fact]
         public void QueryString_AssertSegmentIsCorrect()
         {
-            // Arrange
```
Comment (mikaelm12) [inline test/Microsoft.AspNetCore.Rewrite.Tests/PatternSegments/QueryStringSegmentTests.cs]: Meh, they don't need to be there 😄

File: src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs (hunk 2 above)
Comment (mikaelm12) [inline src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs]: There's actually no `string.IsEmpty` method in C#. It would have to be a string compare to the `string.Empty` field. I think I'll just leave null and empty check 😄

File: test/Microsoft.AspNetCore.Rewrite.Tests/IISUrlRewrite/ServerVariableTests.cs [hunk 4]
```diff
@@ -22,7 +22,7 @@ public class ServerVariableTests
         [InlineData("HTTP_USER_AGENT", "useragent")]
         [InlineData("HTTP_CONNECTION", "connection")]
         [InlineData("HTTP_URL", "/foo")]
-        [InlineData("QUERY_STRING", "?bar=1")]
+        [InlineData("QUERY_STRING", "bar=1")]
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Rewrite.Tests/IISUrlRewrite/ServerVariableTests.cs]: Needs a test for an empty query

File: src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs (hunk 2 above)
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs]: XD kinda surprised by that.

File: test/Microsoft.AspNetCore.Rewrite.Tests/IISUrlRewrite/ServerVariableTests.cs (hunk 4 above)
Comment (jkotalik) [inline test/Microsoft.AspNetCore.Rewrite.Tests/IISUrlRewrite/ServerVariableTests.cs]: How about null?
